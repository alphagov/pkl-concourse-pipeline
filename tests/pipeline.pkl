amends "../src/Pipeline.pkl"

jobs = new {
  new {
    name = "test-job"
    plan = new {
      new GetStep {
        get = "pkl-concourse-pipeline-src"
        trigger = true
      }
      new TaskStep {
        task = "test-task-step"
        config = new {
          platform = "linux"
          image_resource = new {
            type = "registry-image"
            source = new {
              ["repository"] = "governmentdigitalservice/pay-node-runner"
              ["tag"] = "node18"
            }
          }
          run = new {
            path = "node"
            args = new {
              #"echo "hello""#
            }
          }
        }
      }
    }
  }
}

resources = new {
  new {
    name = "pkl-concourse-pipeline-src"
    type = "git"
    icon = "github"
    source = new {
      ["uri"] = "https://github.com/alphagov/pkl-concourse-pipeline"
      ["branch"] = "main"
    }
  }
}

groups = new {
  new {
    name = "pipeline"
    jobs = new {
      "test-job"
    }
  }
}
