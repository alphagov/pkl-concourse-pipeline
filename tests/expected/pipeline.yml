jobs:
- name: test-job
  plan:
  - get: pkl-concourse-pipeline-src
    trigger: true
  - task: test-task-step
    config:
      platform: linux
      image_resource:
        type: registry-image
        source:
          repository: governmentdigitalservice/pay-node-runner
          tag: node18
      run:
        path: node
        args:
        - echo "hello"
resources:
- name: pkl-concourse-pipeline-src
  type: git
  source:
    uri: https://github.com/alphagov/pkl-concourse-pipeline
    branch: main
  icon: github
groups:
- name: pipeline
  jobs:
  - test-job
